File: 1544-make-the-string-great/1544-make-the-string-great.py
class Solution:
    def makeGood(self, s: str) -> str:
        stk = []
        for c in s:
            if stk and abs(ord(stk[-1]) - ord(c)) == 32:
                stk.pop()
            else:
                stk.append(c)
        return "".join(stk)
        
        
        
        
# class Solution:
#     def makeGood(self, s: str) -> str:
#         for i in range(len(s)-1):
#             if abs(ord(s[i]) - ord(s[i+1])) == 32:
#                 return self.makeGood(s[:i]+s[i+2:])
#         return s

File: 1544-make-the-string-great/1544-make-the-string-great.java
class Solution {
    public String makeGood(String s) {
        Stack<Character> stk = new Stack<>();
        for(int i=0; i<s.length(); ++i){
            if(!stk.isEmpty() && (stk.peek() ^ 32) == s.charAt(i))
                stk.pop();
            else
                stk.push(s.charAt(i));
        }
        
        char[] ans = new char[stk.size()];
        
        for(int i=0; i<ans.length; ++i)
            ans[ans.length - i - 1] = stk.pop();
        
        return new String(ans);
    }
}